Makefile

```diff
@@ -88,5 +109,13 @@ push:
 	$(IMAGE_PUSH_CMD) $(IMAGE_TAG)
 	for tag in $(IMAGE_EXTRA_TAGS); do $(IMAGE_PUSH_CMD) $$tag; done
 
+site-build:
+	@mkdir -p docs/vendor/bundle
+	$(SITE_BUILD_CMD) sh -c "bundle install && jekyll build $(JEKYLL_OPTS)"
+
+site-serve:
```

Comment:
I am getting 
```bash
[eduardo@fedora-ws node-feature-discovery-operator]$ make site-build  site-serve  CONTAINER_RUN_CMD="podman run" 
podman run --rm -i -u "`id -u`:`id -g`" -e JEKYLL_ENV=development --volume="$PWD/docs:/srv/jekyll" --volume="$PWD/docs/vendor/bundle:/usr/local/bundle" --network=host jekyll/jekyll:3.8 sh -c "bundle install && jekyll build -d '_site' "
Could not locate Gemfile
make: *** [Makefile:114: site-build] Error 10
```

Reply:
I wasn't able to reproduce this.

Are the mounts working as expected? Selinux??

What do you see when running
```
podman run --rm -i -e JEKYLL_ENV=development --volume="$PWD/docs:/srv/jekyll" --volume="$PWD/docs/vendor/bundle:/usr/local/bundle" --network=host jekyll/jekyll:3.8 ls -l
```
??

However, I encountered problems with user namespaces / uid-gid-mapping and added one patch in an attempt to address that.

Reply:
damm
```bash
[eduardo@fedora-ws node-feature-discovery-operator]$ podman run --rm -i -e JEKYLL_ENV=development --volume="$PWD/docs:/srv/jekyll" --volume="$PWD/docs/vendor/bundle:/usr/local/bundle" --network=host jekyll/jekyll:3.8 ls -l
total 0
ls: can't open '.': Permission denied
```

Reply:
Damn podman (setup) 😠  You don't have permissions to write under `/usr/local/bundle` inside the container which is `$PWD/docs/vendor/bundle:/usr/local/bundle` in the host.

What do you get with:
```
id -u; id -g; podman run --rm jekyll/jekyll:3.8 sh -c "id -u; id -g; cat /proc/self/uid_map; cat /proc/self/gid_map"
```

In all my setups (Fedora 33, openSUSE), I get something like
```
1234
1234
0
0
         0       1234          1
         1     100000      65536
         0       1234          1
         1     100000      65536
```

I.e. root inside the container is mapped to my user id in the host os -> container process is able to manipulate files under the mounted directories

Reply:
https://opensource.com/article/18/12/podman-and-user-namespaces Expected behaviour user namespaces :) Not a bug its a feature.

Reply:
Host and container are running with different SELinux contexts. You're not allowed to just write host content from a container. Does it need to go to `/usr/local/bundle` ? I would suggest to create a directory for the container usage and label it then either with the volumemount option `:Z` or `chcon -t container_file_t`, but I would not change host default directories

Reply:
If you really want to go with the head through the wall a `--privileged` will run the container with `spc_t` type that has access to container and host file system.

Reply:
> Does it need to go to `/usr/local/bundle`

Inside the container it is `/usr/local/bundle` but on the host it is under the current working dir i.e. `$PWD/docs/vendor/bundle`

Reply:
The reason to use containerized build of documentation was supposed to be user convenience 😩 I.e. have fixed user-independent build environment with no need for the user to install any extra docs-related tools on the host. I think we can enable host-based build, too. But in CI (build and verify) I'd use the containerized version

Reply:
Ok than since this is a to the user local directory you can do the following 
```
 -v $PWD/docs/vendor/bundle:/usr/local/bundle:Z 
```
This will label the files correctly to be used inside of the container and then run it with uid=0 to be able to write inside of the container to `/usr/local/bundle`

Reply:
> Ok than since this is a to the user local directory you can do the following
> 
> ```
>  -v $PWD/docs/vendor/bundle:/usr/local/bundle:Z 
> ```

OK, didn't know about that. I added `:Z` to the volume mount options inside Makefile. 

@zvonkok can you try it, what happens with `make site-serve  CONTAINER_RUN_CMD="podman run"`?

Reply:
```
Configuration file: /srv/jekyll/_config.yml
            Source: /srv/jekyll
       Destination: _site
 Incremental build: disabled. Enable with --incremental
      Generating...
      Remote Theme: Using theme rundocs/jekyll-rtd-theme
   GitHub Metadata: No GitHub API authentication could be found. Some fields may be missing or have incorrect data.
fatal: not a git repository (or any parent up to mount point /srv)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
                    done in 7.492 seconds.
 Auto-regeneration: enabled for '/srv/jekyll'
    Server address: http://127.0.0.1:4000
  Server running... press ctrl-c to stop.
```

No errors happily site serving ....

Reply:
Is this the expected output?

Reply:
Yeah, looks good, try browsing to `localhost:4000`

Reply:
Happily browsing, did not check every link but it works for me, :) and on my computer. 

```
Get started · Node Feature Discovery
   Link: canonical
   Link: next
   Node Feature Discovery
   v0.7
   _____________________
   Get started
     * 1. Introduction
     * 2. Quick start
     * 3. Deployment and usage
     * 4. Feature discovery
     * 5. Examples and demos
   Advanced
     * 1. Developer guide
     * 2. Master cmdline reference
     * 3. Worker cmdline reference
   Contributing
   Node Feature Discovery
     * * get-started
     * index.md

     ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

                                                                                                                              Node Feature Discovery

   Welcome to Node Feature Discovery – a Kubernetes add-on for detecting hardware features and system configuration!

   Continue to:

     * Introduction for more details on the project.

     * Quick start for quick step-by-step instructions on how to get NFD running on your cluster.

Quick-start – the short-short version

 $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/node-feature-discovery/v0.7.0/nfd-master.yaml.template
   namespace/node-feature-discovery created
 ...

 $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/node-feature-discovery/v0.7.0/nfd-worker-daemonset.yaml.template
   daemonset.apps/nfd-worker created

 $ kubectl -n node-feature-discovery get all
   NAME                              READY   STATUS    RESTARTS   AGE
   pod/nfd-master-555458dbbc-sxg6w   1/1     Running   0          56s
   pod/nfd-worker-mjg9f              1/1     Running   0          17s
 ...

 $ kubectl get no -o json | jq .items[].metadata.labels
   {
     "beta.kubernetes.io/arch": "amd64",
     "beta.kubernetes.io/os": "linux",
     "feature.node.kubernetes.io/cpu-cpuid.ADX": "true",
     "feature.node.kubernetes.io/cpu-cpuid.AESNI": "true",
 ...
```